pkg/prometheus/promcfg.go

```diff
@@ -4873,6 +4873,17 @@ func (cg *ConfigGenerator) appendNameValidationScheme(cfg yaml.MapSlice, nameVal
 	return cg.WithMinimumVersion("3.0.0").AppendMapItem(cfg, "metric_name_validation_scheme", strings.ToLower(nameValidationSchemeValue))
 }
 
+func (cg *ConfigGenerator) appendNameEscapingScheme(cfg yaml.MapSlice, nameEscapingScheme *monitoringv1.NameEscapingSchemeOptions) yaml.MapSlice {
+	if nameEscapingScheme == nil {
+		return cfg
+	}
+
+	// need to cast it to a string in order to use strings.ToLower() to render the value in the way prometheus expects it
+	nameEscapingSchemeValue := string(*nameEscapingScheme)
```

Comment:
we might need a map[string]string to avoid coupling between CRD and Prometheus values.

Reply:
Thanks @simonpasquier 
Add `ToString()` in CRD
Please re-check it